### DISCUSSION TITLE
Tags on rules that can yield more than one pair

### DISCUSSION THREAD

**Discussion by agatti** [pest-parser/pest#979] [Q&A] [answered]
Hi there!

I'm using pest for the very first time and I'm not sure if I bumped into a bug or not.  So, given this grammar fragment on the latest stable version (2.7.7 with "grammar-extras"):

```pest
tag = {
	// stuff here
}

container = {
	// stuff here
	"{" ~ #tags=tag* ~ "}"
	// more stuff here
}
```

I'd assume that calling `find_tagged("tags")` on a `Pair<Rule::container>` would return all the `Pair<Rule::tag>` that are between brackets.  However what I see is that `find_tagged` returns only the _last_ tag seen by the parser, as if there's just one available pair per tag and it gets overwritten as the parser goes through data.  The grammar seems to be correct as if I remove the `#tags=` bit and do the inner pair iteration myself all the tags are there.

I've read the pest book and I couldn't find anything saying there are limitations on tags attached to rules that may yield more than a pair.  Is this a bug, an undocumented limitation, or am I doing things wrong somewhere?

**Comment by tomtau** [comment]
It looks a bit like a bug; it's due to the fact how the node tagging code is generated with the respect to the parser state changes (it's tagged just once after all those repeated expressions are consumed). I'll see if there's an easy way to fix it; if not, I agree it'll be good to document this limitation.

**Comment by agatti** [comment]
I see, thanks for looking into it.  Do you want me to file a bug to keep track of the issue status?

**Reply by tomtau** [reply]
I opened this issue: https://github.com/pest-parser/pest/issues/982

**Reply by tomtau** [reply, marked as answer]
@agatti this issue should hopefully be fixed in 2.7.8: https://github.com/pest-parser/pest/discussions/986 you can try if it works as expected

**Reply by agatti** [reply]
Yes, it indeed works as expected.  Thanks a lot for the fix!
